python/GafferUI/ColorChooser.py

```diff
@@ -401,7 +402,47 @@ def __drawValue( self, painter ) :
 
 		painter.setBrush( QtGui.QBrush( color ) )
 
-		painter.drawEllipse( QtCore.QPoint( position.x, position.y ), 4.5, 4.5 )
+		size = self.size()
+		if position.x >= 0 and position.y >= 0 and position.x <= size.x and position.y <= size.y :
+			painter.drawEllipse( QtCore.QPoint( position.x, position.y ), 4.5, 4.5 )
+			return
+
+		# Draw a triangle pointing out of the field. Prioritize corners to avoid cutting
+		# off part of the triangle out of the field when one axis is right at 0 or size.
+		#        |\  <-- We don't want this overhang.
+		# -------|->
+		#        |/|
+		#          |
+		#          |
+		if position.x >= size.x and position.y <= 0 :
+			rotation = -45.0  # Triangle pointing to the top-right
+		elif position.x <= 0 and position.y <= 0 :
+			rotation = -135.0  # Top-left
+		elif position.x <= 0 and position.y >= size.y :
+			rotation = -225.0  # Bottom-left
+		elif position.x >= size.x and position.y >= size.y :
+			rotation = -315.0  # Bottom-right
+
+		elif position.x <= size.x and position.x >= 0 and position.y <= 0 :
+			rotation = -90.0  # Top
+		elif position.x <= 0 and position.y >= 0 and position.y <= size.y :
+			rotation = -180.0  # Left
+		elif position.x >= 0 and position.x <= size.x and position.y >= size.y :
+			rotation = -270.0  # Bottom
+		else :
+			rotation = 0.0  # Right
+
+		rightPoints = [ imath.V2f( 0, 0 ), imath.V2f( -4.5, 4.5 ), imath.V2f( -4.5, -4.5 ) ]
```

Comment:
This seems to result in a triangle that is one pixel smaller than the ones in the regular sliders. The slider seems to do it relative to height, which is perhaps inconsistent between platforms? I'd be happy to hardcode the size in both widgets if that helps keep things aligned.

Reply:
I think that was just a miscalculation, it was not exactly the right size for me either. I left the slider code alone and fixed the `_ColorField` drawing in https://github.com/GafferHQ/gaffer/pull/5944/commits/5119e5161619e0b6106443b7a2c968c5a1d03346